cmd/bucket-lifecycle.go

```diff
@@ -583,6 +583,10 @@ func enqueueTransitionImmediate(obj ObjectInfo, src lcEventSrc) {
 	if lc, err := globalLifecycleSys.Get(obj.Bucket); err == nil {
 		switch event := lc.Eval(obj.ToLifecycleOpts()); event.Action {
 		case lifecycle.TransitionAction, lifecycle.TransitionVersionAction:
+			if obj.DeleteMarker || obj.IsDir {
```

Comment:
Can't we just skip 0 byte objects altogether?

Reply:
We could see a new enhancement to skip temporary objects in a separate PR which @krisis is already working on.